### PR TITLE
Update quickstarts and minor docs spruces

### LINKED ISSUE #2225: Deploy FHIR server with CLI will soon be out of date
**Describe the bug**
When using the CLI command to deploy a FHIR server, a warning appears about the command being soon deprecated:

az group deployment create -g $servicename --template-uri https://raw.githubusercontent.com/Microsoft/fhir-server/master/samples/templates/default-azuredeploy.json --parameters serviceName=$servicename

This command is implicitly deprecated because command group 'group deployment' is deprecated and will be removed in a future release. Use 'deployment group' instead.

**FHIR Version?**
all

**Data provider?**
all

**To Reproduce**
1.  follow steps on QuickstartDeployCLI.md

**Expected behavior**
We should update the instructions with the latest CLI command, so we do not get a warning.

**Actual behavior**
A warning appears

### LINKED ISSUE #2326: QuickstartDeployCLI with solutionType=FhirServerSqlServer fails
I was trying to use Sql Server in [QuickstartDeployCLI.md](https://github.com/microsoft/fhir-server/blob/main/docs/QuickstartDeployCLI.md). Encountered a few issues:

1. Examples reference the master branch- project has migrated to the main branch (not a blocker, just a heads up, bunch of issues like this in the repo).
2. Attempting to run the command with the addition of `solutionType=FhirServerSqlServer` results in:
   ```
   az deployment group create --resource-group fhir --template-uri https://raw.githubusercontent.com/Microsoft/fhir-server/main/samples/templates/default-azuredeploy.json --parameters serviceName=$servicename solutionType=FhirServerSqlServer
   {"error":{"code":"InvalidTemplate","message":"Deployment template validation failed: 'The provided value '' for the template parameter 'sqlSchemaAutomaticUpdatesEnabled' at line '163' and column '29' is not valid. The parameter value is not part of the allowed value(s): 'auto,tool'.'.","additionalInfo":[{"type":"TemplateViolation","info":{"lineNumber":163,"linePosition":29,"path":"parameters.sqlSchemaAutomaticUpdatesEnabled.allowedValues"}}]}}
   ```
3. Adding `sqlSchemaAutomaticUpdatesEnabled=auto` results in:
   ```
   az deployment group create --resource-group fhir --template-uri https://raw.githubusercontent.com/Microsoft/fhir-server/main/samples/templates/default-azuredeploy.json --parameters serviceName=$servicename solutionType=FhirServerSqlServer sqlSchemaAutomaticUpdatesEnabled=auto
   {"error":{"code":"InvalidTemplate","message":"Deployment template validation failed: 'The template resource 'Microsoft.Sql/servers/fhir/databases/FHIR' cannot reference itself. Please see https://aka.ms/arm-template-expressions/#reference for usage details.'.","additionalInfo":[{"type":"TemplateViolation","info":{"lineNumber":0,"linePosition":0,"path":""}}]}}
   ```
   image

Is there a way to deploy the project using SQL server?

cc @jdanyow

### PR DESCRIPTION
**Pull request by aidanplunkitt**
## Description
* Update references in documentation from `master` branch to `main`
* Resolve deprecation warnings from `az group deployment` command in quickstarts (2225)
* Add steps for deployment using SQL Store backend, and fix issues related to that (2326)
* Update deployment template references to use the docker template, per the comment on 2326

## Related issues
Fixes #2225, fixes #2326

## Testing
Tested manually by following updated quickstart instructions (used my branch for the updated deployment template url, otherwise SQL Server deployment instructions still fail ala #2326).

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [ ] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [ ] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [ ] CI is green before merge
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Skip (docs)

### CODE REVIEW

File: samples/templates/default-azuredeploy-docker.json [hunk 1]
```diff
@@ -154,7 +154,7 @@
         "sqlSchemaAutomaticUpdatesEnabled": {
             "type": "string",
             "allowedValues":["auto", "tool"],
-            "defaultValue": "[if(equals(parameters('solutionType'),'FhirServerCosmosDB'), 'tool', '')]",
+            "defaultValue": "[if(equals(parameters('solutionType'),'FhirServerCosmosDB'), 'tool', 'auto')]",
```
Comment (LTA-Thinking) [inline samples/templates/default-azuredeploy-docker.json:157]: We are intentionally leaving this field blank. Both the tool and auto settings have drawbacks and we don't want to put one as the default. Leaving it blank forces users to look at the description of the field and make a conscious choice for which they want.

File: samples/templates/default-azuredeploy.json [hunk 2]
```diff
@@ -161,7 +161,7 @@
         "sqlSchemaAutomaticUpdatesEnabled": {
             "type": "string",
             "allowedValues":["auto", "tool"],
-            "defaultValue": "[if(equals(parameters('solutionType'),'FhirServerCosmosDB'), 'tool', '')]",
+            "defaultValue": "[if(equals(parameters('solutionType'),'FhirServerCosmosDB'), 'tool', 'auto')]",
```
Comment (LTA-Thinking) [inline samples/templates/default-azuredeploy.json:164]: Same as above, we want to leave this blank.

**Comment by LTA-Thinking** [comment]
Thank you @aidanplunkitt for these fixes!

File: samples/templates/default-azuredeploy-docker.json (hunk 1 above)
Comment (aidanplunkitt) [reply samples/templates/default-azuredeploy-docker.json:157]: Got it! I'll revert this and add a sqlSchemaAutomaticUpdatesEnabled=auto argument to the sample SQL quickstart commands

Comment (aidanplunkitt) [reply samples/templates/default-azuredeploy-docker.json]: Just added that reversion and manual testing for the new PS and AzCLI quickstarts looks good, so marking this as closed. Please let me know if you have any other feedback on this!

File: samples/templates/default-azuredeploy.json (hunk 2 above)
Comment (aidanplunkitt) [reply samples/templates/default-azuredeploy.json]: Reverted this as well, thanks!

**Review by LTA-Thinking — APPROVED** [review]

**Comment by mikaelweave** [comment]
Looks great @aidanplunkitt!! Thanks for the contribution.

**Pull request merged**